<launch>

  <group ns="left">
    <arg name="arm" default="l"/>
    <arg name="verbose" default="True" />

    <arg name="robot_config" default="$(find hrl_feeding_task)/params/pr2_config_params_l.yaml" />
    <rosparam command="load" file="$(arg robot_config)" />

    <node pkg="pr2_arm_kinematics" type="pr2_arm_kinematics_node" name="pr2_left_arm_kinematics" output="screen">
      <param name="tip_name" value="l_wrist_roll_link" />
      <param name="root_name" value="torso_lift_link" />
    </node>
    <node name="waypoint_generator" pkg="hrl_haptic_mpc" type="waypoint_generator.py" args="-r pr2 -s fabric -a $(arg arm)" output="screen"/>
    <node name="robot_haptic_state" pkg="hrl_haptic_mpc" type="robot_haptic_state_node.py" args="-r pr2 -s fabric -a $(arg arm)" output="screen"/>
    <node name="haptic_mpc" pkg="hrl_haptic_mpc" type="haptic_mpc.py" args="-r pr2 -a $(arg arm) -v $(arg verbose)" output="screen"/>
  </group>

</launch>
